src/main/scala/io/phdata/retirementage/loadgen/LoadGenerator.scala

```diff
@@ -14,11 +18,23 @@ object LoadGenerator {
 
   def generateTables(spark: SparkSession, conf: LoadGeneratorConfig): Unit = {
     // generate fact table
+    val factdf = generateTable(spark, conf.factCount.getOrElse(0), 1)
     // write fact table to disk
+    factdf.write
+      .mode(SaveMode.Overwrite)
+      .parquet(s"${conf.databaseName}.${conf.factName.getOrElse("factLoadTest")}")
     // generate dimension table
+    val dimdf = generateTableFromParent(spark, conf.dimensionCount.getOrElse(0), 3, factdf)
```

Comment:
It will look cleaner if you set the for 'dimensionCount' in `LoadGeneratorConfig`, then you can use the Scallop syntax of getting the value `dimensionCount()` instead of using `get` or `getOrElse`

Reply:
I meant 'set the default in dimensionCount'